Beauty

OlesyaBelochka opened the pull request.

OlesyaBelochka committed:
Increment 1

OlesyaBelochka committed:
Increment 2 (handler)

OlesyaBelochka committed:
Increment 2 (full)

OlesyaBelochka committed:
Increment 3

OlesyaBelochka committed:
Increment 3

OlesyaBelochka committed:
Increment 3 (refactoring)

OlesyaBelochka committed:
Increment 3 (refactoring)

OlesyaBelochka committed:
Increment 3 (add go.sum)

OlesyaBelochka committed:
Increment 3 (changes statuses + add conditions)

OlesyaBelochka committed:
Increment 3 (counter =0)

OlesyaBelochka committed:
Increment 3 (refactor handler & test_handler)

OlesyaBelochka committed:
Increment 3 (change HandleGetMetric)

OlesyaBelochka committed:
Increment 3 (change server main.go = part1)

OlesyaBelochka committed:
Increment 3 (change server main.go = part2)

OlesyaBelochka committed:
Increment 3 (change server main.go = part3)

OlesyaBelochka committed:
Increment 3 (change server main.go = part4)

OlesyaBelochka committed:
Increment 3 (change server main.go = part5)

OlesyaBelochka committed:
Increment 3 (change server main.go = part6)

OlesyaBelochka committed:
Increment 3 (change server main.go = part7)

OlesyaBelochka committed:
Increment 3 (change server main.go = part8)

OlesyaBelochka committed:
Increment 3 (change server main.go = part9)

OlesyaBelochka committed:
Increment 3 (change server main.go = part10)

OlesyaBelochka committed:
Increment 3 (change server main.go = part10)

OlesyaBelochka committed:
Increment 3 (change server main.go = part11)

OlesyaBelochka committed:
Increment 3 (change server main.go = part12)

OlesyaBelochka committed:
Increment 3 (change server main.go = part13)

OlesyaBelochka committed:
Increment 3 (change server main.go = part14)

OlesyaBelochka committed:
Increment 3 (change server main.go = part15)

OlesyaBelochka committed:
Increment 3 (change server main.go = part15)

OlesyaBelochka committed:
Increment 3 (change server main.go = part16)

OlesyaBelochka committed:
Increment 3 (change server main.go = part17)

OlesyaBelochka committed:
Increment 3 (change server main.go = part18)

OlesyaBelochka committed:
Increment 3 (change server main.go = part19)

OlesyaBelochka committed:
Increment 3 (change server main.go = part20)

OlesyaBelochka committed:
Increment 3 (change server main.go = part21)

OlesyaBelochka committed:
Increment 3 (change server main.go = part22)

OlesyaBelochka committed:
Increment 3 (#1 - code review)

OlesyaBelochka committed:
Increment 3 (#1 - code review)

OlesyaBelochka committed:
Increment 3 (#1 - log = true)

OlesyaBelochka committed:
Increment 3 (#1 - log = true)

OlesyaBelochka committed:
Increment 3 (#1 - log = returm old agent)

OlesyaBelochka committed:
Increment 3 (#1 - log = returm old agent=1)

OlesyaBelochka committed:
Increment 3 (#1 - log = returm old agent=2)

OlesyaBelochka committed:
Increment 3 (#2)

OlesyaBelochka committed:
Increment 3 (#3)

OlesyaBelochka committed:
Increment 3 (#4 edd Exit in the sendRequest)

OlesyaBelochka committed:
Increment 3 (#5 check test)

OlesyaBelochka committed:
Increment 3 (#6)

OlesyaBelochka committed:
Increment 3 (#6)

OlesyaBelochka committed:
Increment 3 (#7)

OlesyaBelochka committed:
Increment 4 (#1)

OlesyaBelochka committed:
Increment 4 (#2)

OlesyaBelochka committed:
Increment 4 (#3)

OlesyaBelochka committed:
Increment 5&6 (#1)

OlesyaBelochka committed:
Increment 5&6 (#2)

OlesyaBelochka committed:
Increment 5&6 (#changed agent)

OlesyaBelochka committed:
Increment 5&6 (#changed HandleUpdateMetrics old)

OlesyaBelochka committed:
Increment 5&6 (#changed1 HandleUpdateMetrics old)

OlesyaBelochka committed:
Increment 5&6 (#changed2 Agent)

OlesyaBelochka committed:
Increment 5&6 (#changed3 Agent)

OlesyaBelochka committed:
Increment 5&6 (#changed4)

OlesyaBelochka committed:
Increment 5&6 (#changed4)

OlesyaBelochka committed:
Increment 5&6 (#changed4)

OlesyaBelochka committed:
Increment 5&6 (#changed5)

OlesyaBelochka committed:
Increment 5&6 (#changed6)

OlesyaBelochka committed:
Increment 4 (create brunch test_inc4)

OlesyaBelochka committed:
Increment 4 (create reporter.go itc)

OlesyaBelochka committed:
Increment 4 (#1)

OlesyaBelochka committed:
Increment 4 (#2)

OlesyaBelochka committed:
Increment 4 (#3)

OlesyaBelochka committed:
Increment 4 (#4)

OlesyaBelochka committed:
Increment 4 (#5)

OlesyaBelochka committed:
beauty for run statictest

OlesyaBelochka committed:
#check run staticstet #2

OlesyaBelochka committed:
#check run staticstet #3

OlesyaBelochka committed:
#check run staticstet #4 change printer err

OlesyaBelochka committed:
#check run staticstet #5

OlesyaBelochka committed:
changed with context

OlesyaBelochka committed:
changed run static

OlesyaBelochka committed:
commenr setFlags agent and server

OlesyaBelochka committed:
reset

OlesyaBelochka committed:
reset

OlesyaBelochka committed:
chanche format

OlesyaBelochka committed:
chanche format2

OlesyaBelochka committed:
log

OlesyaBelochka committed:
log2

OlesyaBelochka committed:
changed mistakes 1

OlesyaBelochka committed:
changed mistakes 2 (add HeapInuse, I've lost it)

OlesyaBelochka committed:
changed mistakes 3 (turn on work with file)

OlesyaBelochka committed:
check flags #1

OlesyaBelochka committed:
check flags #1 add log flags

OlesyaBelochka committed:
check flags #3

OlesyaBelochka committed:
check flags #4

OlesyaBelochka committed:
check flags #4

OlesyaBelochka committed:
check flags #5

OlesyaBelochka committed:
check flags #6 setFlags

OlesyaBelochka committed:
check flags #7 correct Println

OlesyaBelochka committed:
(increment 8 compression) test 1

OlesyaBelochka committed:
(increment 7 flag) fix problem in setFlag

OlesyaBelochka committed:
(increment 7 flag) fix problem in setFlag agent

OlesyaBelochka committed:
(increment 7 flag) fix problem in setFlag agent 2

OlesyaBelochka committed:
(increment 7 flag) fix problem in setFlag agent 3

OlesyaBelochka committed:
(increment 7 flag) fix problem in setFlag agent 4

OlesyaBelochka committed:
(increment 7 flag) fix problem in setFlag server 6

OlesyaBelochka committed:
(increment 7 flag) вернула все к комиту днем

OlesyaBelochka committed:
Merge remote-tracking branch 'origin/newAgentWithContext' into newAgentWithContext

# Conflicts:
#	cmd/server/main.go

OlesyaBelochka committed:
(increment 7 flag) странно по прежднему не работает

OlesyaBelochka committed:
(increment 7 flag) странно по прежднему не работает1

OlesyaBelochka committed:
(increment 7 flag) проба 1

OlesyaBelochka committed:
(increment 7 flag) проба 2

OlesyaBelochka committed:
(increment 7 flag) проба 3

OlesyaBelochka committed:
(increment 7 flag) проба 4

OlesyaBelochka committed:
(increment 7 flag) проба 5

OlesyaBelochka committed:
(increment 7 flag) проба 6

OlesyaBelochka committed:
(increment 7 flag) проба 7

OlesyaBelochka committed:
(increment 7 flag) проба 8

OlesyaBelochka committed:
(increment 7 flag) проба 9

OlesyaBelochka committed:
(increment 7 flag) проба 10

OlesyaBelochka committed:
(increment 7 flag) проба 11

OlesyaBelochka committed:
(increment 7 flag) проба 12

OlesyaBelochka committed:
(increment 7 flag) проба 13

OlesyaBelochka committed:
(increment 7 flag) проба 14

OlesyaBelochka committed:
(increment 7 flag) проба 15

OlesyaBelochka committed:
(increment 7 flag) проба 16

OlesyaBelochka committed:
(increment 7 flag) проба 17

OlesyaBelochka committed:
(increment 7 flag) проба 18 ConfigAgent

OlesyaBelochka committed:
(increment 7 flag) проба 18 ConfigAgent

OlesyaBelochka committed:
(increment 8)  change HandleGetMetricJSON

OlesyaBelochka committed:
(increment 8)  change HandleGetMetricJSON w.Header().Set("Accept-Encoding", "gzip")

OlesyaBelochka committed:
(increment 8)  change HandleGetMetricJSON HandleUpdateMetricJSON

OlesyaBelochka committed:
(increment 8)  text/html

OlesyaBelochka committed:
(increment 8)  encoding eccept

OlesyaBelochka committed:
(increment 8)  preplace

OlesyaBelochka committed:
(increment 8)  # 1

OlesyaBelochka committed:
(increment 8)  # 2

OlesyaBelochka committed:
(increment 8)  # 3

OlesyaBelochka committed:
beuty agent and server

OlesyaBelochka committed:
check beauty2

OlesyaBelochka committed:
return updater.go

schetininl commented on .env:
```diff
@@ -0,0 +1,6 @@
+ADDRESS="127.0.0.1:8080"
```
.env файлики - классный инструмент, при этом их не принято хранить в репозиториях

OlesyaBelochka replied:
удалила из репозитория

schetininl commented on cmd/agent/main.go:
```diff
@@ -1,270 +1,40 @@
-	fmt.Println(fullPuth)
-	sendRequest(fullPuth, client)
+var (
+	fАddr               string
```
Названия переменных ничего не говорят о том, что в них содержится

schetininl commented on cmd/agent/main.go:
```diff
@@ -1,270 +1,40 @@
+	go reporters.ReportAgent(ctx)
+	sigEnd := <-osSigChan
+	fmt.Println("Get signal", sigEnd)
+	cancel()
```
cancel можно объявить с помощью defer. Кроме этого, блокировать приложения с помощью osSigChan не самая лучшая идея, так как помимо перечисленных сигналов бывают и другие. 
Блокировать приложения принято в логически основной горутине, например, когда ты запускаешь http server, то там же и происходит блокировка. 
Поэтому, я бы лучше в main сделал примерно так:
`ctx, cancelFn := context.WithCancel(context.Background())
	defer cancelFn()

	osSigChan := make(chan os.Signal, 1)
	signal.Notify(osSigChan, os.Interrupt, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
	go func() {
		<-osSigChan
		logger.Debugf(ctx, "terminating caught")
		cancelFn()
	}()`

OlesyaBelochka replied:
ПОчему, когда я так делаю, у меня клиент завершается? 

	osSigChan := make(chan os.Signal, 1)
	signal.Notify(osSigChan, os.Interrupt, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)

	
	go func() {
		<-osSigChan
		fmt.Println("Get signal")
		cancel()
	}()
вот результат: 

"
2022/05/19 12:23:45 Client started, update and report to IP  127.0.0.1:8080
Address 127.0.0.1:8080, ReportInterval = 10000000000, PollInterval =  2000000000 

Process finished with the exit code 0
"
Вынесла пока

schetininl commented on cmd/server/main.go:
```diff
@@ -1,3 +1,59 @@
+	// зададим встроенные middleware, чтобы улучшить стабильность приложения
+	r.Use(middleware.RequestID)
+	r.Use(middleware.RealIP)
+	//if variables.ShowLog {
```
Мертвый код

schetininl commented on cmd/server/main.go:
```diff
@@ -1,3 +1,59 @@
+	r.Post("/value", handlers.HandleGetMetricJSON)
+
+	if config.ConfS.Address != "" {
+		http.ListenAndServe(config.ConfS.Address, r)
```
Тут забыла проверять error, который может вернуться из ListenAndServe

OlesyaBelochka replied:
+

schetininl commented on internal/storage/srorage.go:
```diff
@@ -0,0 +1,23 @@
+	"runtime"
+)
+
+type Storage interface {
```
fileStorage тоже было бы хорошо под этот интерфейс сделать

OlesyaBelochka replied:
можно я не сейчас? у меня много времени уходит на разработку интерфейсов, а уже надо идти дальше, обещаю чуть позже это реализовать.

schetininl commented on internal/variables/variables.go:
```diff
@@ -0,0 +1,7 @@
+package variables
+
+//const (
```
Мертвый код

schetininl commented on internal/updater/updater.go:
```diff
@@ -0,0 +1,23 @@
+
+func UpdateCountMetric(name string, val variables.Counter) {
+
+	//if variables.ShowFullLog {
```
Мертвый код

OlesyaBelochka replied:
это не мервый, они используются в hendler.go. 
Я перенесла их временно туда, но вообще надо будет переделывать сервер, так же как и агент, через StorageMemory, и в последствии я эти функции уберу.

schetininl commented on internal/updater/updater.go:
```diff
@@ -0,0 +1,23 @@
+)
+
+func UpdateGaugeMetric(name string, val variables.Gauge) {
+	//if variables.ShowFullLog {
```
Мертвый код

OlesyaBelochka replied:
это не мервый, они используются в hendler.go. 
Я перенесла их временно туда, но вообще надо будет переделывать сервер, так же как и агент, через StorageMemory, и в последствии я эти функции уберу.

OlesyaBelochka committed:
changes after code review

OlesyaBelochka committed:
fix autotest #1

OlesyaBelochka committed:
Change reporter.go (use interface)

OlesyaBelochka committed:
Сделала инкременты 9 и 10, отправляю на первое тестирование

OlesyaBelochka committed:
переименовала модуль hash.go  в prhash.go
была ошибка
internal/reporters/reporter.go:9:2: no required module provides package github.com/OlesyaBelochka/My-go-musthave-devops/internal/Prhash; to add it:
	go get github.com/OlesyaBelochka/My-go-musthave-devops/internal/Prhash

OlesyaBelochka committed:
изменила в хеднлерах получение метрик из Мемори.
Дополнила inmemory.go
Реализовала процедуры Get для обеих меток.

OlesyaBelochka committed:
в inmemory.go пепепутала конвертацию для gauge и counter. Поправила, отправляю на тестирование

OlesyaBelochka committed:
нашла ошибку перепутала MGServer v MCServer

OlesyaBelochka committed:
Реализовала инкремент 11, вынесла poll в отдельный файл, В файле мейн сделала чтобы в качестве хранилища бралась либо память, либо БД

OlesyaBelochka committed:
Ошибка при тестах, поправила хендлер, добавила проверку на nil для хранилищь метрик чтобы проходил инкремент 2

OlesyaBelochka committed:
вынесла объявление инмемори в начало,а не после условия если srage, иначе все падало в ошибку при тестах (спринт 4- 9 не проходился)

OlesyaBelochka committed:
добавила понятных комментариев, буду анализировать тесты.

OlesyaBelochka committed:
Реализовала инкремент 12;
Добавила в Storage процедуры SetSlice.
Описала их реализации для всех возможных хранилищ
Реализовала hendler обновляющий пакет меток
Код относящийся к хешу вынесла в отдельную процедуру verificationHash

OlesyaBelochka committed:
Поправила возникающую ошибку при попытке записать большое число.
ERROR: integer out of range (SQLSTATE 22003)
Так же не проходит тест 2, добавила объявление памяти.
Согласно инкременту 12 поменяла чтобы ошибки в процедурах возвращались последними.

OlesyaBelochka committed:
Реализация Инкремента 13, заменила везде fmt.Print, сделала логирование ошибок и информации Info, полученной для анализа выполненных действий.

OlesyaBelochka committed:
поправила функцию генерации хеша.
не проходил 9 инкремент

OlesyaBelochka committed:
Поправила ошибки по statistic test

OlesyaBelochka committed:
Поправила чтобы  hendler возвращал json с хешем если у сервера есть ключ для хеширования

OlesyaBelochka committed:
не прошли инкремент 2 и инкремент 9 поправила кое-что

OlesyaBelochka committed:
еще поправила,скорее все это вызовет ошибку

OlesyaBelochka committed:
Поменяла процедуры получения меток, ранее метки со значением 0 возвращались как ненайденные, а это неверно.
В БД могут храниться метки с нулевыми значениями.

OlesyaBelochka committed:
Поменяла агента как просил Алексей

OlesyaBelochka committed:
Поменяла агента как просил Алексей

OlesyaBelochka committed:
Поменяла агента как просил Алексей исправление static test
